Simplenote/Classes/NoteOptionsViewController.swift

```diff
@@ -0,0 +1,289 @@
+                       footerText: NSLocalizedString("Publish your note to the web and generate a shareable URL", comment: "Note Options Footer: Publish your note to generate a URL"),
+                       rows: rows)
+    }
+
+    /// Configures a section to display our collaboration details
+    fileprivate var collaborationSection: Section {
+        let rows = [
+            Row(style: .Value1,
+                configuration: { (cell: UITableViewCell, row: Row) in
+                    let cell = cell as! Value1TableViewCell
+                    cell.textLabel?.text = NSLocalizedString("Collaborate", comment: "Note Options: Collaborate")
+                    cell.accessibilityLabel = NSLocalizedString("Tap to open collaboration menu", comment: "Accessibility hint on cell that opens collaboration menu")
+                },
+                handler: { [weak self] in
+                    self?.handleCollaborate()
+                }
+            )
+        ]
+        return Section(rows: rows)
+    }
+
+    /// Configures a section to display our trash options
+    fileprivate var trashSection: Section {
+        let rows = [
+            Row(style: .Value1,
+                configuration: { (cell: UITableViewCell, row: Row) in
+                    let cell = cell as! Value1TableViewCell
+                    cell.textLabel?.text = NSLocalizedString("Move to Trash", comment: "Note Options: Move to Trash")
+                    cell.textLabel?.textColor = .simplenoteDestructiveActionColor
+                    cell.accessibilityHint = NSLocalizedString("Tap to move this note to trash", comment: "Accessibility hint on cell that moves a note to trash")
+                },
+                handler: { [weak self] in
+                    self?.handleMoveToTrash()
+                }
+            )
+        ]
+        return Section(rows: rows)
+    }
+
+    // MARK: - Private Nested Classes
+    /// Contains all data required to render a `UITableView` section
+    fileprivate struct Section {
+        /// Optional text to display as standard header text above the `UITableView` section
+        let headerText: String?
+
+        /// Optional text to display as standard footer text below the `UITableView` section
+        let footerText: String?
+
+        /// Any rows to be displayed inside this `UITableView` section
+        let rows: [Row]
+
+        internal init(headerText: String? = nil, footerText: String? = nil, rows: [Row]) {
+            self.headerText = headerText
+            self.footerText = footerText
+            self.rows = rows
+        }
+    }
+
+    /// Contains all the data required to render a row
+    fileprivate struct Row {
+        /// Determines what cell is used to render this row
+        let style: Style
+
+        /// Called to set up this cell. You should do any view configuration here and assign targets to elements such as switches.
+        let configuration: ((UITableViewCell, Row) -> Void)?
```

Comment:
Potentially speaking, `UITableViewCell` could be replaced with generics.

**BUT** potentially speaking, you'd end up with type erasure here and there, and **potentially** speaking, we'd end up with a severe headache 😄

Reply:
Yeah I did briefly start down the generics route then wasn't sure it was the right option for this particular view. I can revisit if you think it's how we want to go about it.

Reply:
Not really, honestly, nearly everytime I pick up Generics, I end up regretting doing so. I think it's all good!